### PR TITLE
Print warning and exit when iOS device is unpaired

### LINKED ISSUE #144447: `flutter run` should detect that targeted iOS devices are unpaired and prompt for how to fix
When I unpair my phone (in phone settings **Developer > Clear Trusted Computers**) and run `flutter devices` or `doctor` I see:
```
Error: iPhone is not paired with your computer. To use iPhone with Xcode, unlock it and choose to trust this computer when prompted. (code -9)
```
However that doesn't seem to be checked during `flutter run` and it totally compiles the app, and eventually fails with a generic "can't launch" message:
```
Could not run build/ios/iphoneos/Runner.app on iPhone.
Try launching Xcode and selecting "Product > Run" to fix the problem:
  open ios/Runner.xcworkspace

Installing and launching...                                         71ms
Error launching application on iPhone.
```
Ideally it would detect the phone is not paired first and give the nicer error message.

```
Flutter (Channel main, 3.20.0-16.0.pre.24, on macOS 14.3.1 23D60 darwin-arm64, locale en-US)
    • Flutter version 3.20.0-16.0.pre.24 on channel main at /Users/magder/Projects/flutter
```

### PR DESCRIPTION
**Pull request by jmagman**
Explicitly handle the case where the iOS device is not paired.  On `flutter run` show an error and bail instead of trying and failing to launch on the device.

On this PR:
```
$ flutter run -d 00008110-0009588C2651401E
'iPhone' is not paired. Open Xcode and trust this computer when prompted.
$
```

Fixes https://github.com/flutter/flutter/issues/144447
Closes https://github.com/flutter/flutter/pull/144095

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: packages/flutter_tools/lib/src/macos/xcdevice.dart [hunk 1]
```diff
@@ -564,6 +561,14 @@ class XCDevice {
           if (code != -10) {
             isConnected = false;
           }
+          // Error: iPhone is not paired with your computer. To use iPhone with Xcode, unlock it and choose to trust this computer when prompted. (code -9)
+          if (errorMessage != null && errorMessage.contains('not paired')) {
+            UsageEvent('device', 'ios-trust-failure', flutterUsage: globals.flutterUsage).send();
+            _analytics.send(Event.appleUsageEvent(workflow: 'device', parameter: 'ios-trust-failure'));
+            if (code == -9) {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/macos/xcdevice.dart:568]: I can't remember if there's a case where we should post the usage event when the code isn't -9, so I kept the old behavior of always posting it if the error message contains "not paired"

File: packages/flutter_tools/lib/src/macos/xcdevice.dart [hunk 2]
```diff
@@ -564,6 +565,10 @@ class XCDevice {
           if (code != -10) {
             isConnected = false;
           }
+          // Error: iPhone is not paired with your computer. To use iPhone with Xcode, unlock it and choose to trust this computer when prompted. (code -9)
+          if (code == -9) {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/macos/xcdevice.dart:569]: Re: the `ios-trust-failure` usage event above: I can't remember if there's a case where the error message is "not paired" but the error code is -9 (I imagine there was or we would have checked this code? Although the error parsing predates the code checking, I believe). So I left the behavior as-is.

**Review by christopherfujino — APPROVED** [review]
LGTM, thanks!

**Review by vashworth — APPROVED** [review]
LGTM!

**Pull request merged**